README.rst

```diff
@@ -3,56 +3,71 @@ openedx-completion-aggregator
+
+The response will include a ``completion`` value between ``0`` and ``1``.
+
+However, what if you want to know the overall % completion of an entire course? ("Alex, you have completed 45% of Introduction to Statistics") Or what if you as an instructor want to get a report of how much of Section 1 every student in a course has completed? Those queries are either not possible or too slow using the APIs built in to the LMS and ``completion``.
+
+This Open edX plugin, ``openedx-completion-aggregator`` watches course activity and asynchronously updates database tables with "aggregate" completion data. "Aggregate" data means completion data summed up over all XBlocks into a course and aggregated at higher levels, like the subsection, section, and course level. The completion aggregator provides a REST API that can provide near-instant answers to queries such as:
+
+* What % complete are each of the courses that I'm enrolled in?
+* What % of each section in Course X have my students completed?
+* What is the average completion % among all enrolled students in a course?
+
+API Details
+-----------
+
+For details about how the completion aggregator's REST APIs can be used, please refer to `the docstrings in views.py <https://github.com/open-craft/openedx-completion-aggregator/blob/master/completion_aggregator/api/v1/views.py#L24>`_.
+
+Installation and Configuration
+------------------------------
+
+openedx-completion-aggregator uses the pluggable django app pattern to ease installation. To use in edx-platform, do the following:
+
+1.  Install the app into your virtualenv::
 
         $ pip install openedx-completion-aggregator
 
-2.  [Optional] You may override the set of registered aggregator block types in
-    your lms.env.json file::
+2.  [Optional] You may override the set of registered aggregator block types in your ``lms.yml`` file::
 
         ...
-        "COMPLETION_AGGREGATOR_BLOCK_TYPES": {
-            "course",
-            "chapter",
-            "subsection",
-            "vertical"
-        },
+        COMPLETION_AGGREGATOR_BLOCK_TYPES:
+            - course
+            - chapter
+            - subsection
+            - vertical
         ...
 
 
-3.  By default, completion is aggregated with each created or updated
-    BlockCompletion.  In most production instances, you will want to calculate
-    aggregations asynchronously.  To enable asynchronous calculation for your
-    installation, set the following in your lms.env.json file::
+3.  By default, completion is aggregated synchronously (with each created or updated BlockCompletion). While that is ideal for development, in most production instances, you will want to calculate aggregations asynchronously for better performance.  To enable asynchronous calculation for your installation, set the following in your ``lms.yml`` file::
```

Comment:
I think it would be nice to clarify the difference between synchronous and asynchronous aggregation. I got a lot queries and there was a lot of confusion about how/when/what data is available and how accurate it is. 

i.e. With synchronous aggregation, each time a student completes a block, it will re-calculate the aggregates. You will always have the freshest results but at a huge performance cost. While with asynchronous aggregation, there will be a delay, since it will only aggregate results at periodic intervals. 
In both cases the single-user, single-course data endpoint will always return the freshest data.

Reply:
Additionally to this it might be worth to make it **very** clear, that the synchronous aggregation can lead to a partial outage by causing deadlocks when users complete the blocks.
Unfortunately this is a real example…

Reply:
Excellent points. I've added a whole new section to the README about synchronous vs. asynchronous. Please check it out.